include/onnxruntime/core/graph/graph_viewer.h

```diff
@@ -210,14 +204,16 @@ class GraphViewer {
 
 #if !defined(ORT_MINIMAL_BUILD)
   // The NodeIndex values of the graph nodes sorted in topological order with priority.
-  std::vector<NodeIndex> nodes_in_topological_order_with_priority_;
-  bool is_priority_order_available_{false};
+  // Using mutable once_flag for lazy initialization.
+  mutable std::vector<NodeIndex> nodes_in_topological_order_with_priority_;
```

Comment:
I'd avoid the idea of `mutable`. Lazy initialization is usually not a good idea since it makes the logics dynamic instead of static for easier debugging. The original logic looks good --- controlling the use of memory-efficient ordering explicitly with explicit flags. May we switch back?

Reply:
There would be a CPU perf regression if we add a new ordering. When GraphViewer is during construction, now 3 topo sort is done, but 2 of them are mostly not used in our graph optimizer paths. There are tons of calls for GraphViewers, that is the reason I wanted to avoid the overhead. Well, if using mutable is something more concerned, I can fallback to sequantial initializations but with known perf degradation when building session.

Reply:
fallbacked.